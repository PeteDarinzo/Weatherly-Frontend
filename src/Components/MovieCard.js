import React from "react";
import { Link } from "react-router-dom";
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { CardActionArea } from '@mui/material';

const MovieCard = ({id, title, img, plot }) => {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardActionArea component={Link} to={`/movies/${id}`}>
        <CardMedia
          component="img"
          height="140"
          image={img}
          alt={title}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="div">
            {title}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {plot}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}

export default MovieCard;
