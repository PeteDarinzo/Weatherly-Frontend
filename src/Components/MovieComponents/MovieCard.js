import React from "react";
import { Link } from "react-router-dom";
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { CardActionArea } from '@mui/material';

const MovieCard = ({ id, title, img }) => {
  return (
    <Card sx={{ height: '100%', backgroundColor: "#2b2b2b" }}>
      <CardActionArea
        component={Link}
        to={`/movies/${id}`}
        sx={{ height: "100%" }}>
        <CardMedia
          component="img"
          image={img}
          alt={title}
        />
        <CardContent>
          <Typography
            variant="subtitle1"
            sx={{ fontSize: "1.2rem" }}
          >
            {title}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}

export default MovieCard;
