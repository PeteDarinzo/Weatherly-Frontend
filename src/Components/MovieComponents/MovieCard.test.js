import MovieCard from "./MovieCard";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

it("renders movie card without crashing", () => {
  render(
    <MemoryRouter>
      <MovieCard id="1" title="test" img="testimg" />
    </MemoryRouter>
  );
});


it("matches the movie card snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <MovieCard id="1" title="test" img="testimg" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
