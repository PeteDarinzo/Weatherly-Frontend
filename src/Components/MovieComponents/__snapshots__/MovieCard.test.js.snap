// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the movie card snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gc2wpw-MuiPaper-root-MuiCard-root"
  >
    <a
      class="MuiButtonBase-root MuiCardActionArea-root css-1pz8km4-MuiButtonBase-root-MuiCardActionArea-root"
      href="/movies/1"
      tabindex="0"
    >
      <img
        alt="test"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        src="testimg"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-1dxvsu6-MuiTypography-root"
        >
          test
        </h6>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
